tests/main/refresh-app-awareness-notify/task.yaml

```
@@ -0,0 +1,48 @@
+summary: Ensure that refresh app-awareness notifications work
+
+details: |
+    Test that notification for continued auto-refreshes are delivered
+
+# Ubuntu 14.04's special version of systemd doesn't have StartTransientUnit API.
+systems: [-ubuntu-14.04-*]
+
+restore: |
+    snap unset system refresh.timer
+
+execute: |
+    # setup test snap and make sure an app keeps it busy
+    
+    snap install test-snapd-sh
+    test-snapd-sh.sh -c 'exec sleep infinity' &
+    PID="$!"
+    # Note that this relies on having different revisions between
+    # test-snapd-sh stable and edge (but other tests require this too)
+    snap switch --edge test-snapd-sh
+
+    # trigger auto-refresh
+    # XXX: do we need this?
+    snap unset system refresh.hold
+    snap set system refresh.timer="$(date -d '1 min' +%H:%M)"
+    i=0
+    while [ $i -lt 12 ]; do
+        sleep 10
+        snap debug ensure-state-soon
+        if journalctl -u snapd | grep "notifying agents about pending refresh for snap test-snapd-sh"; then
+            break
+        fi
+        i=$((i+1))
+    done
+    journalctl -u snapd | MATCH "notifying agents about pending refresh for snap test-snapd-sh"
+
+    # exit wormhole, this should trigger the refresh right away
+    kill "$PID"
+
+    i=0
+    while [ $i -lt 12 ]; do
+        sleep 10
+        if journalctl -u snapd | grep "notifying user client about continued refresh for test-snapd-sh"; then
+            break
+        fi
+        i=$((i+1))
+    done
+    journalctl -u snapd | MATCH "notifying user client about continued refresh for test-snapd-sh"
```

I think this could also be simplified to:

```suggestion
   retry -n 12 --wait 10 sh -c 'journalctl -u snapd | MATCH "notifying user client about continued refresh for test-snapd-sh"'
```